tfjs-backend-webgpu/src/kernels/binary_op_webgpu.ts

```diff
@@ -52,16 +52,38 @@ export class BinaryOpProgram implements WebGPUProgram {
   workGroupSize: [number, number, number] = [16, 1, 1];
 
   constructor(op: string, aShape: number[], bShape: number[]) {
-    this.outputShape = backend_util.assertAndGetBroadcastShape(aShape, bShape);
-    const size = util.sizeFromShape(this.outputShape);
+    if (util.arraysEqual(aShape, bShape) &&
```

Comment:
Would you mind rewriting this slightly so that only what's actually different between the two conditions is included in the if/else statement? I think this will shorten the file and make it easy to see what the optimizations are.

For example `this.outputShape = backend_util.assertAndGetBroadcastShape(aShape, bShape);` will still work for both conditions, could you move it above the if/else? Same with `this.dispatchLayout`. For `computeDispatch`, could you call it outside the if/else, and define the last argument of `computeDispatch` within the if/else? Same pattern for `this.userCode`. 

Could you also do this for `unary_up_webgpu` below?

Reply:
Move most of variables out of if/else and keep `this.userCode`. `this.userCode` is not too complicated. So I prefer not to split part of it out of main body. The advantage is we can clearly see the whole body. Of course, the disadvantage is  `binaryOperation` repeated twice. But it's not a big deal. How do you think?